tests/webtests/ServiceOverviewTest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * Copyright 2018 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Webtests;
+
+use Ramsey\Uuid\Uuid;
+use Symfony\Bundle\FrameworkBundle\Client;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+class ServiceOverviewTest extends WebTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->loadFixtures();
+
+        $this->getAuthorizationService()->setSelectedServiceId(
+            $this->getServiceRepository()->findByName('SURFnet')->getId()
+        );
+    }
+
+    /**
+     * From the services overview page we summarize the available services
+     * for the contact that is currently logged in
+     *
+     * Todo: add tests for the entity form
+     * Todo: The link on the h1 will probably be moved elsewhere
+     */
+    public function test_users_can_use_the_service_overview_page()
+    {
+        $serviceRepository = $this->getServiceRepository();
+        $surfNet = $serviceRepository->findByName('SURFnet');
+        $this->logIn('ROLE_USER', [$surfNet]);
+
+        $crawler = $this->client->request('GET', '/');
+
+        $nodes = $crawler->filter('.card h1');
+        $serviceNode = $nodes->first();
+        $this->assertEquals('SURFnet', $serviceNode->text());
+        $link = $serviceNode->filter('a')->first()->link();
+
+        $this->client->click($link);
+        $this->assertRegExp('#entities$#', $this->client->getResponse()->headers->get('location'));
+    }
+
+    /**
+     * Multiple services can be listed on the page
+     */
+    public function test_multiple_services_can_be_listed()
+    {
+        $serviceRepository = $this->getServiceRepository();
+        $surfNet = $serviceRepository->findByName('SURFnet');
+        $ibuildings = $serviceRepository->findByName('Ibuildings B.V.');
+        $this->logIn('ROLE_USER', [$surfNet, $ibuildings]);
+
+        $crawler = $this->client->request('GET', '/');
+
+        $nodes = $crawler->filter('.card h1');
+
+        $this->assertEquals(2, $nodes->count());
```

Comment:
I personally find it hard to tell what's happening in this code block? I think it would be very valuable to add a small comment.

Reply:
See ca4ed06